app/controllers/users.js

```diff
@@ -1,17 +1,21 @@
-const { createUser, findAndCountAllUsers } = require('../services/users');
+const { createUser, findAndCountAllUsers, createAdminUser } = require('../services/users');
 const { creationParamsMapper } = require('../mappers/users');
 const { paginationParamsMapper } = require('../mappers/pagination_params');
 const { showUserSerializer, usersPageSerializer } = require('../serializers/users');
 const { hashPassword } = require('../helpers/passwords');
 
-exports.createUser = (req, res, next) => {
+const userCreationEndpoint = creator => (req, res, next) => {
   const userBody = creationParamsMapper(req.body);
   return hashPassword(userBody.password)
-    .then(password => createUser({ ...userBody, password }))
+    .then(password => creator({ ...userBody, password }))
     .then(user => res.status(201).send(showUserSerializer(user)))
     .catch(next);
 };
```

Comment:
Great refactor!
Two comments:
- I think it would be cleaner if the params where: (req, res, next, creator)
- Could we change the name for something without endpoint in it?

Reply:
1 - I think not because when I then need to use like:
```js
exports.createUser = (req, res, next) => userCreationEndpoint(req, res, next, createUser)
```
On the `creator => (req, res, next)` way I think it is more clearly that creator is a dependency. I can't use this function without a creator,
And when i give a creator to this function it returns a middleware-like function.

2- Agree. I'll think a name.